data-analytics/dataflow-python-examples/google/data_enrichment.py

```diff
@@ -0,0 +1,190 @@
+                i += 1
+
+            return row
+
+
+def run(argv=None):
+    """The main function which creates the pipeline and runs it"""
+    # This object helps us parse command line arguments
+    parser = argparse.ArgumentParser()
+    # Here we add some specific command line arguments we expect.   Specifically
+    # we have the input file to load and the output table to write to.
+    parser.add_argument(
+        '--input', dest='input', required=False,
+        help='Input file to read.  This can be a local file or '
+             'a file in a Google Storage Bucket',
+        # this example file contains a total of only 10 lines.
+        # Useful for quickly debugging on a small set of data
+        default='gs://python-dataflow-example/data_files/head_usa_names.csv')
+    # The output defaults to the lake dataset in your BigQuery project.  You'll have
+    # to create the lake dataset yourself using this command:
+    # bq mk lake
+    parser.add_argument('--output', dest='output', required=False,
+                        help='Output BQ table to write results to.',
+                        default='lake.usa_names_enriched')
+
+    # Parse arguments from the command line
+    known_args, pipeline_args = parser.parse_known_args(argv)
+
+    # DataIngestion is a class we built in this script to hold the logic for
+    # transforming the file into a BigQuery table
+    data_ingestion = DataIngestion()
+
+    # Initiate the pipeline using the pipeline arguments passed in from the
+    # command line.  This includes information like where Dataflow should store
+    #  temp files, and what the project id is
+    p = beam.Pipeline(options=PipelineOptions(pipeline_args))
+    schema = parse_table_schema_from_json(data_ingestion.schema_str)
+    # pipeline = beam.Pipeline(options=PipelineOptions(pipeline_args))
+
+    # This function adds in a full state name by looking up the
+    # full name in the short_to_long_name_map.  The short_to_long_name_map
+    # comes from a read from BigQuery in the next few lines
+    def add_full_state_name(row, short_to_long_name_map):
+        row['state_full_name'] = short_to_long_name_map[row['state']]
+        return row
+
+    # This is a second source of data.  The source is from BigQuery.
+    # This will come into our pipeline a "Side Input".
+    state_abbreviations = (
+        p
+        | 'Read from BigQuery' >> beam.io.Read(
+            beam.io.BigQuerySource(query='select state_name, state_abbreviation'
+                                         ' from '
+                                         '`python-dataflow-example.example_data.state_abbreviations`'
+                                   , use_standard_sql=True))
+        # We must create a python "Tuple" of key to value pairs here in order to
+        # use the data as a side input.  Dataflow will use the keys to distribute the
+        # work to the correct worker.
+        | 'Abbreviation to Full Name' >> beam.Map(
```

Comment:
DataFlow op names should conform to standard CamelCase, e.g. AbbreviationToFullName